### PR TITLE
Character Counter

### PR DESCRIPTION
**Pull request by ourmaninamsterdam**
Displays remaining number of characters permitted in an input based on a given maximum. A warning is displayed if said maximum is exceeded. 
Integrated into the refactored article edit form.

TP#3663

### CODE REVIEW

File: spec/javascripts/tests/CharacterCounter_spec.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    it('should find and cache at least one indicator element', function() {
+      this.component.init();
+      expect(this.component.$indicators.length).to.be.at.least(1);
+    });
+
+    it('should update the character indicator with the number of inputted characters', function() {
+      var str = 'foo';
+      this.component.$el.val(str);
+      this.component.init();
+      expect(parseInt(this.component.$indicators.eq(0).text(),10)).to.eq(this.component.maxChars - str.length);
```
Comment (tomas-stefano) [inline spec/javascripts/tests/CharacterCounter_spec.js]: Can we break the actual and the expected in two variables?

**Comment by dougdroper** [comment]
:+1: after rebase

**Comment by ourmaninamsterdam** [comment]
@moneyadviceservice/frontend Anymore thumbs?

**Comment by tomas-stefano** [comment]
:+1:

**Pull request merged**